index2.js

```diff
@@ -0,0 +1,269 @@
+    }else{
+      console.log('There is no keypad on this side.');
+      getInput();
+    }
+  }
+
+  inventory(){
+    if(player.playerInventory.length < 1){
+      console.log("You have nothing.")
+      return getInput();
+    } else {
+      player.playerInventory.forEach(function(element) {
+        console.log(element.inventoryText);
+      });
+      return getInput();
+    }
+  }
+  move(newState){
+    if(Object.keys(roomLookupTable).includes(newState) && this.currentState !== "mainSt"){
+    console.log(this.currentState);
+      enterState(newState, this.currentState);
+    console.log(this.currentState.description)
+    } else {
+      console.log(`I can't move to ${newState}`)
+    }
+   getInput();
+  }
+
+  async xyzzy(){
+    let newRoom = await ask("Which room would you like to go to mainSt or foyer?")
+    if(Object.keys(roomLookupTable).includes(newRoom)){
+      this.currentState = newRoom;
+      console.log(roomLookupTable[player.currentState].name);
+      console.log(roomLookupTable[player.currentState].description);
+    }else{
+          console.log(`That isn't a choice. Please try again`);
+          return xyzzy();
+  }
+   return getInput();
+  }
+}
+
+//Items
+const paper = new Item(
+"Seven Days", 
+"Some kind of newspaper with bands you've never heard of.", 
+true,
+`You pick up the paper and leaf through it looking for comics and ignoring the articles, just like everybody else does.`,
+"You are carrying Seven Days, Vermonts Alt-Weekly",
+"You flip through the pages. Nothing catches your eye."
+);
+
+const sign = new Item(
+  "Sign", 
+  `Welcome to Burlington Code Academy! Come on 
+up to the third floor. If the door is locked, use the code 
+12345`,
+  false,
+  "That would be selfish. How will other students find their way?",
+  "",
+  "",
+);
+
+const door = new Item(
+  "door",
+  "182 Main St",
+  false,
+  "It's locked and I don't know how you'd carry it.",
+  "",
+  "The door is locked. There is a keypad on the door handle."
+);
+
+//Rooms
+let mainSt = new Room(
+  "182 Main St.",
+  `You are standing on Main Street between Church and South Winooski.
+There is a door here. A keypad sits on the handle.
+On the door is a handwritten sign.`,
+  [sign, door],
+  ['mainSt', 'foyer']
+  );
+
+let foyer = new Room(
+'182 Main St. - Foyer',
+`You are in a foyer. Or maybe it's an antechamber. Or a 
+vestibule. Or an entryway. Or an atrium. Or a narthex.
+But let's forget all that fancy flatlander vocabulary,
+and just call it a foyer. In Vermont, this is pronounced
+"FO-ee-yurr".
+A copy of Seven Days lies in a corner.`,
+[paper],
+['street', 'roomThree'],
+);
+
+  //Do we need a lookup table for [current].description to work? Marshall mentioned that you can't have the first 
+  //key be a variable unless you use a lookup table to convert the strimovng to an object by naming itself
+  const roomLookupTable = {
+    mainSt: mainSt,
+    foyer: foyer,
+    "street": mainSt
+    //roomThree: roomThree
```

Comment:
You may want to add in more alternate names for your rooms to extend the allowable input for your `move` function. In that same vein you may want another look up table that'll allow alternate names for your actions e.g. 
```js
const altNames = {
  move: ['move', 'go', 'run'],
  take: ['take', 'pick up', 'grab'] 
}
```